base_classes/NXsource.nxdl.xml

```diff
@@ -163,55 +184,118 @@
 		<doc>For storage rings, the current at the end of the most recent injection.</doc>
 		<attribute name="time" type="NX_DATE_TIME"><doc>date and time of the most recent injection.</doc></attribute>
 	</field>
-        <group name="geometry" type="NXgeometry" deprecated="Use the field `depends_on` and :ref:`NXtransformations` to position the source and NXoff_geometry to describe its shape instead">
+	<field name="wavelength" type="NX_FLOAT" units="NX_WAVELENGTH">
+		<doc>
+				The wavelength of the radiation emitted by the source.
+		</doc>
+	</field>
+	<field name="pulse_energy" type="NX_FLOAT" units="NX_ENERGY">
+		<doc>
+				For pulsed sources, the energy of a single pulse.
+		</doc>
+	</field>
+	<field name="peak_power" type="NX_FLOAT" units="NX_POWER">
+		<doc>
+				For pulsed sources, the pulse energy divided
+				by the pulse duration
+		</doc>
+	</field>
+	<field name="anode_material">
+		<doc>
+				Material of the anode (for X-ray tubes).
+		</doc>
+	</field>
+	<field name="filament_current" type="NX_FLOAT" units="NX_CURRENT">
+		<doc>
+				Filament current (for X-ray tubes).
+		</doc>
+	</field>
+	<field name="emission_current" type="NX_FLOAT" units="NX_CURRENT">
+		<doc>
+				Emission current of the generated beam.
+		</doc>
+	</field>
+	<field name="gas_pressure" type="NX_FLOAT" units="NX_PRESSURE">
+		<doc>
+				Gas pressure inside ionization source.
+		</doc>
+	</field>
```

Comment:
Move these into sub-classes (wavelength through gas_pressure)

Reply:
IMO, there is no really good to move these into sub-classes. Some of these fields are for X-ray tubes, some for more ionization sources, some for pulsed source. `wavelength` on the other hand is certainly a concept for many different kinds of sources.

I could create 3 or 4 classes here (each inheriting) from `NXsource`, like `NXxray_tube` and `NXpulsed_source`, each of which takes one or two of these new concepts. But what about an pulsed X-ray source, that would be again the merger of the two.

To me, it makes more sense to keep them in the `NXsource` base class.

Reply:
I'm ok with your reasoning here but technically it's not what we voted on.  Let's review how to proceed on Monday.

Reply:
Resolved in the telco to keep as is